### PR TITLE
delivery and taxi websockets

### PR DESCRIPTION
**Pull request by ahmednasserzaza**
- add new attributes that needed to trip model.
- add status to trip (pending - approved - arrived - finished).
- add flag for trip in which app is created (taxi ride - delivery order).
- implement websocket for every type of trip(taxi ride - delivery order).
- remove unneccessary delete trip endpoint.
- refactor all update status endpoints and add endpoints that doesn't existed.
- call all created websockets and needed endpoints in ApiGateway 

socket_success
socket_success_2
socket_error_2
socket_error_response

apiGateway_add_delivery_trip
apiGateway_approve_delivery_trip
apiGateway_socket_recieve_delivery_trip
apiGateway_socket_recieve_taxi_trip
apiGateway_trip_arrived
apiGateway_trip_finished
service_add_trip
service_approve_trip
service_finish_trip
service_socket_recieve_delivery_trip
service_socket_recieve_taxi_trip

### CODE REVIEW

**Review by mosayed01 — APPROVED** [review]

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/data/service/RestaurantService.kt [hunk 1]
```diff
@@ -331,7 +331,7 @@ class RestaurantService(
         )
     }
 
-    suspend fun restaurantOrders(restaurantId: String, languageCode: String): Flow<OrderDto> {
+    suspend fun restaurantOrders(restaurantId: String): Flow<OrderDto> {
```
Comment (AzizaHelmy) [inline api_gateway/src/main/kotlin/org/thechance/api_gateway/data/service/RestaurantService.kt:334]: name of the function should be verb

**Review by Youssef-Ramadan-Y — APPROVED** [review]

**Review by devsadeq — APPROVED** [review]

**Review by mosayed01 — APPROVED** [review]

**Review by ay0o5h — APPROVED** [review]

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/data/model/taxi/DeliveryTrackingResponse.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package org.thechance.api_gateway.data.model.taxi
+
+import kotlinx.serialization.Serializable
+import org.thechance.api_gateway.data.model.LocationDto
+
+@Serializable
+data class DeliveryTrackingResponse(
```
Comment (AzizaHelmy) [inline api_gateway/src/main/kotlin/org/thechance/api_gateway/data/model/taxi/DeliveryTrackingResponse.kt:7]: add serilization annotation

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/data/model/taxi/DeliveryTripResponse.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package org.thechance.api_gateway.data.model.taxi
+
+import kotlinx.serialization.Serializable
+import org.thechance.api_gateway.data.model.LocationDto
+
+@Serializable
+data class DeliveryTripResponse(
```
Comment (AzizaHelmy) [inline api_gateway/src/main/kotlin/org/thechance/api_gateway/data/model/taxi/DeliveryTripResponse.kt:7]: add serilization annotation

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/data/model/taxi/RideTrackingResponse.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package org.thechance.api_gateway.data.model.taxi
+
+import kotlinx.serialization.Serializable
+import org.thechance.api_gateway.data.model.LocationDto
+
+@Serializable
+data class RideTrackingResponse(
```
Comment (AzizaHelmy) [inline api_gateway/src/main/kotlin/org/thechance/api_gateway/data/model/taxi/RideTrackingResponse.kt:7]: add serilization annotation

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/data/model/taxi/TaxiTripResponse.kt [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package org.thechance.api_gateway.data.model.taxi
+
+import kotlinx.serialization.Serializable
+import org.thechance.api_gateway.data.model.LocationDto
+
+@Serializable
+data class TaxiTripResponse(
```
Comment (AzizaHelmy) [inline api_gateway/src/main/kotlin/org/thechance/api_gateway/data/model/taxi/TaxiTripResponse.kt:7]: add serilization annotation

**Review by AzizaHelmy — APPROVED** [review]

**Pull request merged**